spec/models/users_spec.rb

```diff
@@ -12,6 +12,164 @@
   describe "validations" do
     it { should validate_presence_of(:username) }
     it { should validate_presence_of(:email) }
+  end
+
```

Comment:
I think most of these tests aren't actually testing the methods. Allow x to receive y and return z just bypasses the actual method execution. I think we should push it for now since some of these queries are written for code that doesn't exist yet, and it'll give us a roadmap for testing later on.